packages/hardhat-core/src/types/runtime.ts

```diff
@@ -183,7 +187,8 @@ export interface TasksMap {
 
 export type RunTaskFunction = (
   name: string,
-  taskArguments?: TaskArguments
+  taskArguments?: TaskArguments,
+  subtaskArguments?: { [subtaskName: string]: TaskArguments }
```

Comment:
Shouldn't this be

```suggestion
  subtaskArguments?: SubtaskArguments
```

?

Reply:
Good catch. It looks better.